index.jsx

```diff
@@ -69,24 +72,22 @@ class MapInteraction extends Component {
 
   static get defaultProps() {
     return {
-      initialX: 0,
-      initialY: 0,
       minScale: 0.05,
       maxScale: 3,
       showControls: false
     };
   }
 
-
   constructor(props) {
     super(props);
+    const { scale, defaultScale, translation, defaultTranslation, minScale, maxScale } = props;
+
+    const desiredScale = !isNaN(scale) ? scale :
```

Comment:
I know it isn't as pretty, but please avoid nested ternarys

```js
    let desiredScale;
    if (!isNaN(scale)) {
      desiredScale = scale;
    } else if (!isNaN(defaultScale)) {
      desiredScale = defaultScale;
    } else {
      desiredScale = 1;
    }
```

I know you don't want the dependency on something like lodash, so you could also write a helper util to find the values.

```js
const find = (values, fn) => {
  for (const value of values) {
    if (fn(value)) return value;
  }

  return null;
}

const scales = [scale, defaultScale, 1];
const desiredScale = find(scales, (v) => !isNaN(v))
```